### PR TITLE
feat: let org admins change usernames

### LINKED ISSUE #13383: [CAL-3010] Org admins should be able to change usernames
org admins should be able to change usernames here:

CleanShot 2024-01-24 at 12 53 44@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3010](https://linear.app/calcom/issue/CAL-3010/org-admins-should-be-able-to-change-usernames)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
TODO

- [x] Fix Avatar update bugs

havent tested but could work, unless the API needs more permission to change username, not sure

CleanShot 2024-01-24 at 12 59 34@2x

### CODE REVIEW

**Review by keithwillcode — DISMISSED** [review]
Type checks are failing

**Comment by PeerRich** [comment]
> Type checks are failing

likely more failing, havent tested locally at all yet 🙏

**Comment by Ryukemeister** [comment]
> Type checks are failing

should fix the type checks now

File: packages/features/users/components/UserTable/EditSheet/EditUserForm.tsx [hunk 1]
```diff
@@ -157,6 +160,7 @@ export function EditForm({
         </div>
       </div>
       <div className="mt-6 flex flex-col space-y-3">
+        <TextField label={t("username")} {...form.register("username")} />
```
Comment (Udit-takkar) [inline packages/features/users/components/UserTable/EditSheet/EditUserForm.tsx:163]: Do we need to check if username is available or not here (like we do in other pages)?

**Review by Udit-takkar — COMMENTED** [review]
packages/trpc/server/routers/viewer/organizations/updateUser.handler.ts doesn't update username 

Screenshot 2024-01-29 at 5 18 42 PM

File: packages/features/users/components/UserTable/EditSheet/EditUserForm.tsx (hunk 1 above)
Comment (PeerRich) [reply packages/features/users/components/UserTable/EditSheet/EditUserForm.tsx:163]: but only within the organisation

**Comment by Udit-takkar** [comment]
Username Updation

https://github.com/calcom/cal.com/assets/53316345/7c11fe71-81dc-4309-a457-8b968ecc88c4

**Review by sean-brydon — DISMISSED** [review]
CleanShot 2024-02-19 at 08 44 00


I tried updating this test user "test-example" to "pro" but got a unique constraint error 

No users in this org have this name either

**Comment by Udit-takkar** [comment]
> 
> I tried updating this test user "test-example" to "pro" but got a unique constraint error
> 
> No users in this org have this name either

Okay let me try to replicate this bug.

**Comment by Udit-takkar** [comment]
@sean-brydon  Works fine for me. There is user with 'pro' username in the DB but not in the org

https://github.com/calcom/cal.com/assets/53316345/bda95149-37b2-46ed-8b9d-77562d258b72

File: packages/trpc/server/routers/viewer/organizations/updateUser.handler.ts [hunk 2]
```diff
@@ -74,30 +79,58 @@ export const updateUserHandler = async ({ ctx, input }: UpdateUserOptions) => {
...
   if (!requestedMember)
     throw new TRPCError({ code: "UNAUTHORIZED", message: "User does not belong to your organization" });
 
-  let avatar = input.avatar;
-  if (input.avatar) {
-    avatar = await resizeBase64Image(input.avatar);
+  const hasUsernameUpdated = input.username !== requestedMember.user.username;
+
+  if (input.username && hasUsernameUpdated && user.profile.organization?.slug) {
+    const checkRegularUsernameRes = await checkRegularUsername(
```
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/organizations/updateUser.handler.ts:96]: NIT: I noticed that `checkRegularUsername` is used only within an org context. Should we rename this function so it's more obvious when reading the function name?

**Review by joeauyeung — APPROVED** [review]
This is working as expected. Left a small nit

**Pull request merged**